rbi/documentation/versioned_annotations.md

```diff
@@ -0,0 +1,70 @@
+# Versioned RBI Annotations
+
+Many gems simultaneously maintain more than one version, often with different external APIs. If the gem's RBI
+annotation file does not match the version being used in your project, it can result in misleading type checking
+errors that slow down development.
+
+Starting in version 0.1.10, the [rbi gem](https://github.com/Shopify/rbi/) supports adding gem version comments
+to RBI annotations, allowing us to specify the gem versions that include specific methods and constants.
+[Tapioca](https://github.com/Shopify/tapioca) version 0.15.0 and later will strip out any parts of the annotation
+file that are not relevant to the current gem version when pulling annotations into a project.
+
+## Syntax
+
+To use this feature, add a comment in the following format directly above a method or constant:
+
+```ruby
+# @version > 0.2.0
+sig { void }
+def foo; end
+```
+
+The comment must start with a space, and then the string `@version`, followed by an [operator](#operators) and
+a version number. Version numbers must be compatible with Ruby's
+[`Gem::Version` specification](https://ruby-doc.org/current/stdlibs/rubygems/Gem/Version.html).
+
+Any method or constant that does not have a version annotation will be considered part of all versions.
+
+## Operators
```

Comment:
Can we add a link to the official documentation of these operators instead of listing them here? https://ruby-doc.org/3.3.4/stdlibs/rubygems/Gem/Version.html